### PR TITLE
Add ability to unit test Jinja2 macros with JSON specification

### PR DESCRIPTION
**Pull request by willbarton**
This PR adds utilities for unit testing Jinja2 macros based on JSON specifications and includes an example specification in `tests/`. The README should explain everything, if it doesn't I should change it so that it does.
 
**Additions**
 
- An abstract `MacroTestCase` Python class
- A factory function to take a test specified in a JSON file and create the appropriate Python class
- A test runner for JSON specification files
- A README that attempts to explain the process.
  
**Testing**
 
From within the `_tests/macro_testing` directory:

- Install the requirements (which should already be installed in a sheer/cfgov-refresh virtualenv): `pip install -r requirements.txt`
- Run the test runner: `python runner.py`
 
**Review**
 
- @anselmbradford 
- @sebworks 
- @jimmynotjim 

**Notes**

I presume that the new flapjack branch is the correct place for this.

### CODE REVIEW

File: _tests/macro_testing/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+## Writing Unit Tests for Jinja2 Macros
+
+There are two ways that unit tests for Jinja2 macros can be written,
+in Python and in a JSON file.
+
+### JSON Test Specifications
+
+To reduce the amount of boilerplate Python that needs to be written for
+macro unit tests, unit tests can be writen in JSON.
```
Comment (cfarm) [inline _tests/macro_testing/README.md]: typo = should be "written"

**Comment by jimmynotjim** [comment]
Created a test for the`event_preview()` macro and it looks like you've worked out all the kinks. I think once this is squashed it'll be good to go :+1:

**Comment by jimmynotjim** [comment]
Love it, thanks for creating this one so fast :shipit:

**Comment by willbarton** [comment]
So the question becomes whether to merge this into cfgov-refresh, or go ahead and make it its own library in its own repo that cfgov-refresh just uses. I'm not sure what the best option is right now... I know long term it should live by itself, and maybe support Django templates and maybe Genshi (does anyone still use Genshi?).

**Comment by jimmynotjim** [comment]
I like the idea of a stand-alone library and I still vote for MacroPolo as it's name

**Comment by sebworks** [comment]
Definitely should be stand-alone. MacroPolo is genius.

**Comment by willbarton** [comment]
Sounds good. I'll get MacroPolo into a less-dependent-on-cfgov-refresh condition and externalize it.

**Comment by willbarton** [comment]
Closing. We'll split this off into MacroPolo and I'll create a new PR with a runner and tests based on that.

**Pull request closed without merging**